portal/test-helpers/test-data/bank-report-period-schedule.ts

```diff
@@ -0,0 +1,16 @@
+import { BankReportPeriodSchedule } from '@ukef/dtfs2-common';
+
+export const aMonthlyBankReportPeriodSchedule = (): BankReportPeriodSchedule => [
```

Comment:
I'm not so sure about this test helper - I thought the `aThing()` pattern was good for getting a base object which can have it's properties overwritten or modified to match your test case whilst still preserving everything else you would expect it to have. In this case, you would never want to modify anything on the return value of this function, so it doesn't fit that criteria (with the caveat that this criteria could just be something my brain has come up with from our current usage of this pattern).

It's also a bit of a semantic thing in the `aMonthlyBankReportPeriodSchedule` uses the indefinite article, but there is only one monthly bank report period schedule. In this context it can also mean "a monthly bank report period schedule *object*" , which is fine with respect to the indefinite article, so if it is meant in this way then I think it's fine.

This is mainly a concern to me because of our current usage of this pattern, as it may lead some unsuspecting dev to write something like
```typescript
const scheduleWithJanuaryToMarch: BankReportPeriodSchedule = [
  ...aMonthlyBankReportPeriodSchedule(),
  { startMonth: 1, endMonth: 3 },
];
```
which wouldn't work as they expect. I doubt someone would end up doing this so I don't feel strongly about it, but what do you think?

Reply:
I actually think the `aThing` helpers are really useful for the tests where we don't want to change anything, just gives us what we need straight up, with zero chance of cross-test interference.
And yeah for me it is the "object"-ness of this that justifies the grammar, I do agreed does feel slightly weirder than some of the other uses.
That example usage would actually be quite a nice way for making an invalid schedule to test the schedule validator for example, and a case where if instead someone accidentally edited a variable we used across tests to achieve it would cause lots of other tests to fail!

Reply:
That all makes sense to me :)